src/dstack/_internal/core/models/backends/oci.py

```diff
@@ -0,0 +1,64 @@
+from pathlib import Path
+
+import oci
+from pydantic import Field
+from typing_extensions import Annotated, Any, List, Literal, Mapping, Optional, Union
+
+from dstack._internal.core.models.backends.base import ConfigMultiElement
+from dstack._internal.core.models.common import CoreModel
+
+
+class OCIConfigInfo(CoreModel):
+    type: Literal["oci"] = "oci"
+    regions: Optional[List[str]] = None
+
+
+class OCIClientCreds(CoreModel):
+    type: Annotated[Literal["client"], Field(description="The type of credentials")] = "client"
+    user: Annotated[str, Field(description="User OCID")]
+    tenancy: Annotated[str, Field(description="Tenancy OCID")]
+    key_file: Annotated[Path, Field(description="Path to the PEM key")]
+    fingerprint: Annotated[str, Field(description="Key fingerprint")]
+    region: Annotated[
+        str, Field(description="Name or key of any region the tenancy is subscribed to")
+    ]
+
+    def to_client_config(self) -> Mapping[str, Any]:
+        return self.dict(exclude={"type"})
+
+
+class OCIDefaultCreds(CoreModel):
+    type: Annotated[Literal["default"], Field(description="The type of credentials")] = "default"
+
+    def to_client_config(self) -> Mapping[str, Any]:
+        return oci.config.from_file()
```

Comment:
I recommend you removing any logic from models. Any logic on top of models can be implemented as functions in core/backend/oci. E.g. https://github.com/dstackai/dstack/blob/64056913d694ed439d0d59cebfe3598683ab56fc/src/dstack/_internal/core/backends/gcp/auth.py#L19